runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4302,13 +4302,13 @@ class INTERPRETER_CLASS
 		I_32 type = CPE_TYPE_UNUSABLE;
 		/* This native is only called for the boot loader, so vmRef is guaranteed to be initialized */
 		J9ClassLoader *classLoader = J9VMJAVALANGCLASSLOADER_VMREF(_currentThread, classLoaderObject);
-		/* Bounds check the parameter */
+		/* Bounds check the parameter. No need to acquire cpEntriesMutex as the cp entries can only grow for bootstrap class path. */
 		if ((cpIndex >= 0) && (cpIndex < (I_32)classLoader->classPathEntryCount)) {
 			/* Check the flags of the translation data struct */
 			J9TranslationBufferSet *translationData = _vm->dynamicLoadBuffers;
 			if (NULL != translationData) {
 				/* Initialize the class path entry */
-				J9ClassPathEntry *cpEntry = classLoader->classPathEntries + cpIndex;
+				J9ClassPathEntry *cpEntry = classLoader->classPathEntries[cpIndex];
```

Comment:
cpEntriesMutex is required for this.

Reply:
Is the muxtex needed for one address read ?

Reply:
Problem is it's not one address but two. Read classLoader->classPathEntries and then index from that to read the entry pointer.